extensions/vertx-redis/runtime/src/main/java/io/quarkus/vertx/redis/runtime/RedisConfig.java

```diff
@@ -0,0 +1,45 @@
+package io.quarkus.vertx.redis.runtime;
+
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.Set;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.vertx.redis.client.RedisClientType;
+
+@ConfigRoot(phase = ConfigPhase.RUN_TIME)
+public class RedisConfig {
```

Comment:
I think if you use the connection string format I think all this class is not needed.

Reply:
Thanks for the review. We'll still need this class as it is the configuration root class. 

Quick question: Does `ssl` work with the new api? Do you have an example how I can set it up?